src/eth_provider/database/types/transaction.rs

```diff
@@ -24,57 +25,37 @@ impl From<Transaction> for StoredTransaction {
 }
 
 #[cfg(any(test, feature = "arbitrary", feature = "testing"))]
-impl<'a> arbitrary::Arbitrary<'a> for StoredTransaction {
-    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
-        let transaction = TransactionSigned::arbitrary(u)?;
+impl<'a> StoredTransaction {
+    pub fn arbitrary_with_optional_fields(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let transaction = Self::arbitrary(u)?.tx;
```

Comment:
`Transaction::arbitrary` instead of `Self::arbitrary(u)?.tx;`?

Reply:
Fixed